linkerd/dns/src/lib.rs

```diff
@@ -101,24 +126,79 @@ impl Resolver {
         let (result_tx, rx) = oneshot::channel();
         self.tx.send(ResolveRequest {
             name,
-            result_tx,
             span,
+            lookup_type: LookupType::Ip(result_tx),
         })?;
         let ips = rx.await??;
         Ok(ips)
     }
 
-    pub fn resolve_one_ip(
+    async fn lookup_service(&self, name: Name, span: Span) -> Result<SrvLookup, Error> {
+        let (result_tx, rx) = oneshot::channel();
+        self.tx.send(ResolveRequest {
+            name,
+            span,
+            lookup_type: LookupType::Service(result_tx),
+        })?;
+        let srvs = rx.await??;
+        Ok(srvs)
+    }
+
+    // XXX We need to convert the SRV records to an IP addr manually,
+    // because of: https://github.com/bluejekyll/trust-dns/issues/872
+    // Here we rely in on the fact that the first label of the SRV
+    // record's target will be the ip of the pod delimited by dashes
+    // instead of dots. We can alternatively do another lookup
+    // on the pod's DNS but it seems unnecessary since the pod's
+    // ip is in the target of the SRV record.
+    fn srv_to_socket_addr(srv: &rdata::SRV) -> Result<net::SocketAddr, Error> {
+        let first_label = srv
+            .target()
+            .iter()
+            .next()
+            .ok_or_else(|| Error::InvalidSRVRecord(srv.clone()))?;
+
+        std::str::from_utf8(first_label)
+            .expect("should be utf8")
+            .replace("-", ".")
+            .parse::<std::net::IpAddr>()
+            .map(|ip| net::SocketAddr::new(ip, srv.port()))
+            .map_err(|_| Error::InvalidSRVRecord(srv.clone()))
+    }
+
+    pub fn resolve_service_ips(
```

Comment:
```suggestion
    pub fn resolve_service_addrs(
```

This looks up addresses (i.e. with port), not just IPs.